### DISCUSSION TITLE
How do you debug the library?

### DISCUSSION THREAD

**Discussion by musicstashall** [ALTaleX531/TranslucentFlyouts#138] [General]
How do you debug the library? After loading, VS does not control the DLL and debugging is not possible. I want to look at the values in the variables and I can't find a way to look into the process.

**Comment by ALTaleX531** [comment]
As I said, you really lack the knowledge and experience to develop Windows desktop apps.
In short, you can use VS's attach to process feature and set breakpoints on a process that has TF loaded so that you can view the execution details of the code.
Instead of asking me, Google is a better teacher for you. I don't understand what features you want to implement yourself even you know nothing about the code.

**Reply by musicstashall** [reply]
That's what I do, I set breakpoints. Before that, I reread a bunch of forums and instructions. Why are you reproaching me? If you have a student in front of you, then you have the opportunity to share your experience. In addition, I have had a project related to yours since 2017, and I am familiar with this topic.

**Reply by ALTaleX531** [reply]
As I said, you don't have any experience with Windows desktop development nor are you familiar with C++, I can't teach you everything from scratch. It's like a student who can't even do the most basic addition and instead asks me about calculus, it's ridiculous.
launch_win32.bat will load the dll for you into most processes, you just need to pick a proper application to attach it to, and after making sure that the compiled version and the code version are the same, you can set a breakpoint, if it is a release version then the code has been optimized a lot during compilation, which makes it basically impossible to accurately hit a breakpoint. For development, you should compile with the debug option.
I've tried to make it as clear as possible how to compile and debug the project, if you still don't get it you should consider it a priority to learn what you're missing. If you're really familiar with the topic, then you should have a lot of experience with dll debugging, which is contrary to the fact.

**Reply by musicstashall** [reply]
Thank you. I figured it out on my own. I did not expect that it was necessary to connect to a third-party process, I have not yet met such an implementation. 

Once, when I was a schoolboy, my classmate came to me and asked me to teach him radio modeling, and since childhood I have been an amateur radio operator. I gave him a task so that he would learn the elements of the circuits, then we put together a radio receiver with him. Many years later, I found out that he had become a radio engineer. 

Yes, I am not familiar with C++, and the last time I wrote in the Studio was in 2010 in the language VB.NET and I've never written a DLL. But I'm learning and I've already mastered a lot. In practice, I master languages in one year. And this development topic is familiar to me, as I said, I have a [project](https://autoit-script.ru/threads/black-window-manager-1-0-5-514-w8-1-w10-x86-x64.25976/) that creates a transparent background for the entire Windows interface, but it was written in autoit, where the implementation of all functions was completely different (We must expect). Google: BlackWindowManager - this is my job and is quite widely known. I am familiar with all the api functions that you use, I even found an error in your project - an incorrect constant value ACCENT_ENABLE_MODERN_ACRYLIC_RECIPE, must be equal to 1. And I have long wanted to write my project in a more correct form, something like yours. And finally, I find such an example on github and get to know the developer. Please be a little more patient with me. I'm far from a teenager anymore, I'm 49 years old. If you don't mind helping me in the future, I would even compensate you moderately financially. GitHub was created just so that we can share our experience and that knowledge multiplies exponentially. I don't think you could be against such a phenomenon))

**Reply by ALTaleX531** [reply]
Unfortunately, however, the value of `ACCENT_ENABLE_MODERN_ACRYLIC_RECIPE` is not a bug, but rather you don't understand how it behaves, and if you wish to have a deep understanding of what's really going on you'll need to decompile `udwm.dll`. From simply being the `SetWindowCompoisitionAttribute` caller, to someone who fully understands how Windows handles it internally, you still have a long way to go. All of this code is a product of my hobby, I never intend to make a profit on it, and I actually use Python for my actual development.
I can't answer all your questions on the internet every day, I have my own work to do, you'll have to learn to search for the answers yourself, as I have done in the past, and it's often more effective than asking someone. In the end, I can only feel pity for what you have experienced and wish you the best of luck.
